aten/src/ATen/KinetoPluginAPI.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <c10/core/DeviceType.h>
+#include <c10/macros/Macros.h>
+
+#include <libkineto.h>
+
+namespace at::kineto_plugin {
+
+using PluginFactory = libkineto::ChildActivityProfilerFactory;
+void SetKinetoPluginRegister(DeviceType t, PluginFactory f);
```

Comment:
But, we should expose this API to developer, right?
```suggestion
TORCH_API using PluginFactory = libkineto::ChildActivityProfilerFactory;
TORCH_API void SetKinetoPluginRegister(DeviceType t, PluginFactory f);
```

Reply:
We plan to let user use **REGISTER_KINETO_PLUGIN** to register their own profiler.

Reply:
By design, line #23 will prevent to overrider the existing register. Therefore, the SetKinetoPluginRegister must be exposed to provide user to overrider the existing one.

Reply:
> By design, line [pytorch#23](https://github.com/pytorch/pytorch/issues/23) will prevent to overrider the existing register. Therefore, the SetKinetoPluginRegister must be exposed to provide user to overrider the existing one.

The words is not correct enough. Anyway, let's expose both macro and func to users.

Reply:
> > By design, line [pytorch#23](https://github.com/pytorch/pytorch/issues/23) will prevent to overrider the existing register. Therefore, the SetKinetoPluginRegister must be exposed to provide user to overrider the existing one.
> 
> The words is not correct enough. Anyway, let's expose both macro and func to users.

ok